packages/ui/src/components/_utils/analytics.ts

```diff
@@ -4,21 +4,34 @@ import React from 'react'
 
 import { version } from '@open-condo/ui/package.json'
 
+interface IAnalyticsProps<K> {
+    location: string
+    component: K
+}
+
 type ComponentSpecificClickEventProps = {
     Banner: { title: string }
     Button: { value: string, type: string }
     'Typography.Link': { value: string, href?: string }
 }
 
-type CommonAnalyticsProps<K> = {
-    event: 'click',
-    location: string,
-    component: K
+type ComponentSpecificCheckEventProps = {
+    Radio: { value: string }
+    Checkbox: { value: string }
 }
 
-type AnalyticsClickData<K extends keyof ComponentSpecificClickEventProps> = CommonAnalyticsProps<K> & ComponentSpecificClickEventProps[K]
+interface CommonAnalyticsClickProps<K> extends IAnalyticsProps<K> {
+    event: 'click'
+}
+
+interface CommonAnalyticsCheckboxProps<K> extends IAnalyticsProps<K> {
+    event: 'checkbox' | 'radio'
```

Comment:
Why not just `check`?
IMO component prop should be responsible for this logic....

Reply:
That's related to the original realisation of "analytics naming".
<img width="709" alt="image" src="https://user-images.githubusercontent.com/25844927/214830301-8b7d7410-f1e9-48ae-be8c-fbaafbb3bef5.png">

Reply:
Where `TrackingEventType` is looks like that:
<img width="280" alt="image" src="https://user-images.githubusercontent.com/25844927/214830586-347520ea-ccab-4f6b-b068-5b89f6c14158.png">

Reply:
It was done this way because local components didn't know type of itself. IMHO it's little dirty way to direct pass type of the component to the tracking context